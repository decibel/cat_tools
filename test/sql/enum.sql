\set ECHO none

\i test/pgxntool/setup.sql

\set test_enum enum_range_srf_test_enum

SELECT plan(3);

SELECT lives_ok(
  $$CREATE TYPE $$ || :'test_enum' || $$ AS ENUM( 'ZZZ Label 1', 'Label 2' )$$
  , 'Create test enum'
);

-- Tests both functions
SELECT results_eq(
  format( 'SELECT cat_tools.enum_range_srf( %L )', :'test_enum' )
  , $$VALUES ( 'ZZZ Label 1' ), ( 'Label 2' )$$
);

SELECT lives_ok(
  'DROP TYPE ' || :'test_enum'
  , 'Drop test enum'
);

\i test/pgxntool/finish.sql

-- vi: expandtab ts=2 sw=2
