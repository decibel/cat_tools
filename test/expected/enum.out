\set ECHO none
1..3
ok 1 - Create test enum
ok 2
ok 3 - Drop test enum
# TRANSACTION INTENTIONALLY LEFT OPEN!
